components/resolved_ts/src/cmd.rs

```diff
@@ -0,0 +1,288 @@
+                    let err_header = response.mut_header().take_error();
+                    Some(ChangeLog::Error(err_header))
+                }
+            })
+            .filter_map(|v| v)
+            .collect()
+    }
+
+    pub fn encode_rows(requests: Vec<Request>, is_one_pc: bool) -> Vec<ChangeRow> {
+        let mut change_logs: HashMap<Key, ChangeRow> = HashMap::default();
+        let mut pending_default: HashMap<Key, Vec<u8>> = HashMap::default();
+
+        for mut req in requests {
+            if req.get_cmd_type() != CmdType::Put {
+                // Do not log delete requests because they are issued by GC
+                // frequently.
+                if req.get_cmd_type() != CmdType::Delete {
+                    debug!(
+                        "skip other command";
+                        "command" => ?req,
+                    );
+                }
+                continue;
+            }
+            let mut put = req.take_put();
+            match put.cf.as_str() {
+                "write" => match decode_write(put.get_key(), put.get_value()) {
+                    Some(write) => {
+                        // TODO: handle gc_fence here.
+                        let key = Key::from_encoded(put.take_key());
+                        let commit_ts = key.decode_ts().unwrap();
+                        let key = key.truncate_ts().unwrap();
+                        if is_one_pc {
+                            change_logs.insert(
+                                key.clone(),
+                                ChangeRow::OnePc {
+                                    commit_ts,
+                                    write,
+                                    value: pending_default.remove(&key),
+                                    key,
+                                },
+                            );
+                        } else {
+                            let commit_ts = if write.write_type == WriteType::Rollback {
+                                None
+                            } else {
+                                Some(commit_ts)
+                            };
+                            change_logs.insert(
+                                key.clone(),
+                                ChangeRow::Commit {
+                                    key,
+                                    commit_ts,
+                                    write,
+                                },
+                            );
+                        }
+                    }
+                    None => continue,
+                },
+                "lock" => match decode_lock(put.get_key(), put.get_value()) {
+                    Some(lock) => {
+                        let key = Key::from_encoded(put.take_key());
+                        let value = pending_default.remove(&key);
+                        let l = change_logs
+                            .insert(key.clone(), ChangeRow::Prewrite { key, lock, value });
```

Comment:
The lock may contain a `short_value`, should we set the `ChangeRow::Prewrite::value` here in this case?

Reply:
Not necessary. Just notice to handle it 🤔.